generators/profilegen/src/resources/common/configurable.rb

```diff
@@ -50,7 +50,16 @@ def reset!(options = {})
         instance_variable_set(:"@#{key}", options.fetch(key, default_value))
       end
 
-      default_value = get_credentials(self.tenant_id, self.client_id, self.client_secret, self.active_directory_settings)
+      fail ArgumentError, 'tenant_id is nil' if self.tenant_id.nil?
+      fail ArgumentError, 'client_id is nil' if self.client_id.nil?
+      fail ArgumentError, 'client_secret is nil' if self.client_secret.nil?
+      fail ArgumentError, 'subscription_id is nil' if self.subscription_id.nil?
+      fail ArgumentError, 'active_directory_settings is nil' if self.active_directory_settings.nil?
```

Comment:
could this be nil at this point? this is not something we're asking the user to pass in, right?

Reply:
Technically, yes. They will not be nil. I just wanted to be cautious that one of our users intentionally passing nil value for active_directory_settings in the options hash.